halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/deploy/sizing/ComponentSizingEditCommand.java

```diff
@@ -0,0 +1,139 @@
+    )
+    private String requestsMemory;
+
+    @Parameter(
+            names = "--pod-limits-cpu",
+            description = "Sets the cpu limit for the container running the spinnaker service, as well as any sidecar containers (e.g. the monitoring daemon). Example: 1."
+    )
+    private String limitsCpu;
+
+    @Parameter(
+            names = "--pod-limits-memory",
+            description = "Sets the memory limit for the container running the spinnaker service, as well as any sidecar containers (e.g. the monitoring daemon). Example: 1Gi."
+    )
+    private String limitsMemory;
+
+    public ComponentSizingEditCommand(String serviceName) {
+        this.serviceName = serviceName;
+    }
+
+    @Override
+    protected String getShortDescription() {
+        return "Edit the component sizing for service " + getServiceName();
+    }
+
+    @Override
+    protected void executeThis() {
+        String currentDeployment = getCurrentDeployment();
+        String serviceName = getServiceName();
+        DeploymentEnvironment deploymentEnvironment = new OperationHandler<DeploymentEnvironment>()
+                .setFailureMesssage("Failed to get component sizing for service " + serviceName + ".")
+                .setOperation(Daemon.getDeploymentEnvironment(currentDeployment, false))
+                .get();
+
+        CustomSizing customSizing = deploymentEnvironment.getCustomSizing();
+        int originalHash = customSizing.hashCode();
+
+        customSizing = edit(customSizing);
+
+        if (originalHash == customSizing.hashCode()) {
+            AnsiUi.failure("No changes supplied.");
+            return;
+        }
+
+        new OperationHandler<Void>()
+                .setFailureMesssage("Failed to edit component sizing for " + serviceName + ".")
+                .setSuccessMessage("Successfully edited component sizing for service " + serviceName + ".")
+                .setOperation(Daemon.setDeploymentEnvironment(currentDeployment, !noValidate, deploymentEnvironment))
+                .get();
+    }
+
+    private CustomSizing edit(CustomSizing customSizing) {
+        Map serviceSizing = customSizing.computeIfAbsent(spinService(), (k) -> new HashMap<>());
+        edit(serviceSizing);
+        return customSizing;
+    }
+
+    private void edit(Map serviceSizing) {
+        putIfNotNull(serviceSizing, "replicas", replicas);
+
+        Map limits = (Map) serviceSizing.computeIfAbsent("limits", (k) -> new HashMap<>());
+        putIfNotNull(limits, "cpu", limitsCpu);
+        putIfNotNull(limits, "memory", limitsMemory);
+
+        Map requests = (Map) serviceSizing.computeIfAbsent("requests", (k) -> new HashMap<>());
+        putIfNotNull(requests, "cpu", requestsCpu);
+        putIfNotNull(requests, "memory", requestsMemory);
+    }
+
+    private void putIfNotNull(Map map, String key, Object value) {
```

Comment:
computeIfAbsent refers to the key being absent. putIfNotNull checks that the value is null, so they are quite distinct. 

If we use computeIfAbsent and can imagine a scenario where you want to increase the number of replicas from 2 to 3:
computeIfAbsent(replicas, 2)
computeIfAbsent(replicas, 3)

The second call is ignored, because the replicas key exists and the value will continue to be 2, which is not what we want.